tools/ncf.py

```diff
@@ -154,6 +156,9 @@ def parse_bundlefile_metadata(content, bundle_type):
         # tag "parameter" may be multi-valued
         if tag == "parameter":
           parameters.append({'name': match.group(3), 'description': match.group(4)})
+        if tag == "parameter_constraint":
+          constraint = json.loads("{" + match.group(4)+ "}")
```

Comment:
Do you expect a list of key values or a list of elements here ? 
Did you write the constraint format somewhere ?